src/prefect/deployments.py

```diff
@@ -80,7 +80,12 @@ async def run_deployment(
 
     parameters = parameters or {}
 
-    deployment = await client.read_deployment_by_name(name)
+    deployment_id = None
+    try:
+        deployment_id = UUID(name_or_id)
+        deployment = await client.read_deployment(deployment_id=deployment_id)
+    except ValueError:
+        deployment = await client.read_deployment_by_name(name_or_id)
```

Comment:
It'd be better to call `read_deployment` outside of the `try/except` block. 

1. If `read_deployment` throws a `ValueError` this code will incorrectly try again by name which should be confusing to debug.
2. If `read_deployment_by_name` throws an error we'll get a weird traceback where an exception occurred while handling an exception.

The `try/except` should just be scoped to the attempt to parse the name into a UUID.